### PR TITLE
Feature/add args algo

### PR DESCRIPTION
**Pull request by sach1095**

### CODE REVIEW

File: src/Parse.cpp [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
 #include <filesystem>
 
-Parse::Parse(char **av) {
+Parse::Parse(char **inputArgs) {
 	this->_isSolvable = true;
+	this->_fileName = false;
 	this->_algo = ASTAR;
 	this->_heuristics = MANHATTAN;
-	ParseAV(av);
+	ParseAV(inputArgs);
```
Comment (abourbou) [inline src/Parse.cpp]: ParseAV is not clear, what do you think of ParseInput ?

File: src/Parse.cpp [hunk 2]
```diff
@@ -125,69 +126,103 @@ void Parse::showParsedContent() {
...
-	std::vector<int> poss;
-	if (idx % s > 0) poss.push_back(idx - 1);
-	if (idx % s < s - 1) poss.push_back(idx + 1);
-	if (idx / s > 0) poss.push_back(idx - s);
-	if (idx / s < s - 1) poss.push_back(idx + s);
+void swapEmptyTile(std::vector<int>& puzzle, int puzzleSize) {
+	int emptyTileIndex = std::distance(puzzle.begin(), std::find(puzzle.begin(), puzzle.end(), 0));
+	std::vector<int> possibleSwaps;
+
+	// Vérifier les mouvements possibles et les ajouter à possibleSwaps
```
Comment (abourbou) [inline src/Parse.cpp]: Change in english please

File: src/Parse.cpp [hunk 3]
```diff
@@ -125,69 +126,103 @@ void Parse::showParsedContent() {
...
+	// Échanger la tuile vide avec la tuile choisie
+	std::swap(puzzle[emptyTileIndex], puzzle[swapWithIndex]);
+}
 
-	std::swap(p[idx], p[swi]);
+std::vector<int> makeGoal(int puzzleSize) {
+    int totalTiles = puzzleSize * puzzleSize;
+    std::vector<int> puzzle;
+
+    // Remplir le puzzle avec des nombres de 1 à puzzleSize*puzzleSize - 1
```
Comment (abourbou) [inline src/Parse.cpp]: English pls

File: src/Parse.cpp [hunk 4]
```diff
@@ -252,28 +280,33 @@ void Parse::ParseAV(char **av){
...
+			if (args.empty())
+				throw CustomError("Error: No given argument for --heuristic");
 			if (args.front() == "manhattan")
 				this->_heuristics = MANHATTAN;
-			else if (args.front() == "other")
-				this->_heuristics = OTHER;
-			else if (args.front() == "othertwo")
-				this->_heuristics = OTHERTWO;
+			else if (args.front() == "linear")
+				this->_heuristics = LINEAR_CONFLICT;
```
Comment (abourbou) [inline src/Parse.cpp:328]: You should make _heuristics a ptr to heuristic type and put directly the function here

File: include/Enum.hpp [hunk 5]
```diff
@@ -10,9 +10,8 @@ enum Algorithm {
 
 enum Heuristics {
 	MANHATTAN = 0,
-	// TODO a implementer
-	OTHER = 1,
-	OTHERTWO = 2,
+	LINEAR_CONFLICT = 1,
```
Comment (abourbou) [inline include/Enum.hpp:13]: Finally this enum is useless, just replace it by the function directly

File: src/Parse.cpp [hunk 6]
```diff
@@ -125,69 +126,103 @@ void Parse::showParsedContent() {
...
-}
 
-void swapEmpty(std::vector<int>& p, int s) {
-	int idx = std::distance(p.begin(), std::find(p.begin(), p.end(), 0));
-	std::vector<int> poss;
-	if (idx % s > 0) poss.push_back(idx - 1);
-	if (idx % s < s - 1) poss.push_back(idx + 1);
-	if (idx / s > 0) poss.push_back(idx - s);
-	if (idx / s < s - 1) poss.push_back(idx + s);
+void swapEmptyTile(std::vector<int>& puzzle, int puzzleSize) {
```
Comment (abourbou) [inline src/Parse.cpp]: You should create a separate file for the generator of puzzle

File: src/Parse.cpp (hunk 1 above)
Comment (sach1095) [reply src/Parse.cpp]: Outdated

File: src/Parse.cpp [hunk 7]
```diff
@@ -1,39 +1,69 @@
...
 #include "Custom_Error.hpp"
+#include "GeneratorMap.cpp"
 #include <filesystem>
 
-Parse::Parse(char **av) {
+Parse::Parse(char **inputArgs)
+{
 	this->_isSolvable = true;
-	ParseAV(av);
+	this->_fileName = false;
```
Comment (abourbou) [inline src/Parse.cpp:9]: Les arguments par default devrait etre dans le .hpp, pas dans le constructeur

File: Makefile [hunk 8]
```diff
@@ -58,13 +58,17 @@ clean	:
...
+	@ rm -rf solutionVisualiser.html
 	@ printf "$(CYAN)'$(NAME)'$(RESET) and all .o has been $(RED)deleted. 🗑️\n$(RESET)"
 
 re		:	fclean all
 
 run : all
 	@ ./$(NAME) --size 3
 
+visu : all
+	@ ./$(NAME) --size 3 --visualiser
```
Comment (abourbou) [inline Makefile:70]: visualiser => viewer
visualisateur de solution => solution viewer

**Review by abourbou — APPROVED** [review]

**Pull request merged**